sync/class.jetpack-sync-module-themes.php

```diff
@@ -11,6 +11,8 @@ public function init_listeners( $callable ) {
 		add_action( 'upgrader_process_complete', array( $this, 'check_upgrader'), 10, 2 );
 		add_action( 'jetpack_installed_theme', $callable, 10, 2 );
 		add_action( 'jetpack_updated_theme', $callable, 10, 2 );
+		add_action( 'delete_site_transient_update_themes', array($this, 'detect_theme_deletion') );
```

Comment:
I haven't looked at how the internals of deleting themes work, but why it is only hooked to `delete_site_transient` couldn't happen on `update_site_transient`?